app/models/wallet-connect-client/wallet-connect-client.ts

```diff
@@ -241,6 +245,48 @@ export const WalletConnectClientModel = types
             }
             break
           }
+
+          case "likerId_login": {
+            try {
+              const [chainId, loginMessage] = payload.params
+              const { bech32Address }: { bech32Address: string } = yield self.env.authCoreAPI.getWalletConnectGetKeyResponse(
+                chainId,
+                { name: self.currentUser.likerID }
+              )
+              const ts = Date.now();
+              let memo = JSON.stringify({
+                ts,
+                likeWallet: bech32Address,
+              })
+              memo = [`${loginMessage}:`, memo].join(' ');
+              const {
+                signed: message,
+                signature: { signature, pub_key: publicKey },
+              }: {
+                signed: any,
+                signature: any,
+              } = yield self.env.authCoreAPI.signAmino({
+                // eslint-disable-next-line @typescript-eslint/camelcase
+                chain_id: chainId,
+                memo,
+                msgs: [],
+                fee: { gas: '1', amount: [{ denom: 'nanolike', amount: '0' }] },
```

Comment:
Can we format this message in frontend, and just call signAmino for wallet?
This would allow frontend to update the challenge in the future, and might also allow easier integration with other wc wallet

Reply:
We can perform special handling if we catch a certain string like "login" in memo though

Reply:
I think the problem is that the current challenge requires frontend to acquire the wallet address to compose the message for signing, and we just want a single WC request call to do that